man/fi_av.3.md

```diff
@@ -117,6 +119,10 @@ int fi_av_lookup_auth_key(struct fid_av *av, fi_addr_t addr,
 : On input, specifies size of auth_key buffer.  On output, stores number
   of bytes written to auth_key buffer.
 
+*user_id*
+: For address vectors configured with FI_AV_USER_ID, this defines the
+  the user ID to be associated with future inserts. NULL is valid.
```

Comment:
This is assigning a generic user ID across all insertions.  I was thinking of something like this:
```
fi_set_user_id(*av, fi_addr_t addr, fi_addr_t user_id)
```

Reply:
Ah. So the order of operations would be:

1. fi_av_insert() -> output is fi_addr_t addr
2. fi_av_set_user_id() -> Update addr from step 1 with user defined ID

Reply:
Correct.

Reply:
That race already exists without a new call.  A CQ event could be generated prior to fi_av_insert() being called.  In that case the fi_addr_t will be FI_ADDR_NOTAVAIL.  In the new model, the CQ event could be generated prior to calling fi_av_set_user_id().  The fi_addr_t value would also be FI_ADDR_NOTAVAIL.  It's the same conceptual race, and there's no reasonable way for the app to determine if the CQ event was generated prior to calling fi_av_insert() or immediately after calling fi_av_insert() but before fi_av_set_user_id().

Reply:
>In the new model, the CQ event could be generated prior to calling fi_av_set_user_id(). The fi_addr_t value would also be FI_ADDR_NOTAVAIL. 

Ah yes. I missed this part. If FI_ADDR_NOTAVAIL is returned, all is good (same race as today).